sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1963,6 +1966,14 @@ class Analyzer(override val catalogManager: CatalogManager)
     override def apply(plan: LogicalPlan): LogicalPlan = {
       val externalFunctionNameSet = new mutable.HashSet[FunctionIdentifier]()
       plan.resolveExpressions {
+        case f @ UnresolvedFunction(NonSessionCatalogAndIdentifier(catalog, name), _, _, _, _) =>
```

Comment:
Why is this done here and not in analysis checks?

Reply:
You mean in `CheckAnalysis`? to my understanding that is done at the end after we've run all the rules, while doing it here allows us to do some sanity checks and detect analysis errors earlier.